### PR TITLE
S/add oi oishares to events

### PR DESCRIPTION
**Pull request by magnetto90**

### CODE REVIEW

File: contracts/OverlayV1Market.sol [hunk 1]
```diff
@@ -253,7 +272,11 @@ contract OverlayV1Market is IOverlayV1Market, Pausable {
         }
 
         // emit build event
-        emit Build(msg.sender, positionId_, oi, debt, isLong, price);
+        if (isLong) {
+            emit Build(msg.sender, positionId_, oi, debt, isLong, price, oiLong, oiLongShares);
+        } else {
+            emit Build(msg.sender, positionId_, oi, debt, isLong, price, oiShort, oiShortShares);
+        }
```
Comment (EperezOk) [inline contracts/OverlayV1Market.sol:279]: I would replace these 5 lines with:
```js
emit Build(msg.sender, positionId_, oi, debt, isLong, price, oiLong, isLong ? oiLongShares : oiShortShares);
```
I think it should work.

File: contracts/OverlayV1Market.sol [hunk 2]
```diff
@@ -359,7 +383,27 @@ contract OverlayV1Market is IOverlayV1Market, Pausable {
         }
 
         // emit unwind event
-        emit Unwind(msg.sender, positionId, fraction, int256(value) - int256(cost), price);
+        if (pos.isLong) {
```
Comment (EperezOk) [inline contracts/OverlayV1Market.sol:386]: Same recommendation as in the `Build` event.

File: contracts/OverlayV1Market.sol [hunk 3]
```diff
@@ -448,13 +493,27 @@ contract OverlayV1Market is IOverlayV1Market, Pausable {
...
+            emit Liquidate(
+                msg.sender,
+                owner,
+                positionId,
+                int256(value) - int256(cost) - int256(marginToBurn),
+                price,
+                oiShort,
+                oiShortShares
+            );
+        }
```
Comment (EperezOk) [inline contracts/OverlayV1Market.sol:516]: Same recommendation as in the build event.

**Review by EperezOk — CHANGES_REQUESTED** [review]

File: contracts/OverlayV1Market.sol (hunk 1 above)
Comment (EperezOk) [reply contracts/OverlayV1Market.sol:279]: You are right about replacing `oiLong` as well.

Regarding what is better, I think what I suggested improves readability, and having one extra comparison is negligible.

File: contracts/OverlayV1Market.sol [hunk 4]
```diff
@@ -383,27 +388,17 @@ contract OverlayV1Market is IOverlayV1Market, Pausable {
...
-                msg.sender,
-                positionId,
-                fraction,
-                int256(value) - int256(cost),
-                price,
-                oiShort,
-                oiShortShares
-            );
-        }
+        bool isLong = pos.isLong;
```
Comment (EperezOk) [inline contracts/OverlayV1Market.sol:391]: No need to create a separate `isLong` variable, we can use `pos.isLong` directly (it's already loaded in memory).

File: contracts/OverlayV1Market.sol [hunk 5]
```diff
@@ -493,27 +488,17 @@ contract OverlayV1Market is IOverlayV1Market, Pausable {
...
-                msg.sender,
-                owner,
-                positionId,
-                int256(value) - int256(cost) - int256(marginToBurn),
-                price,
-                oiShort,
-                oiShortShares
-            );
-        }
+        bool isLong = pos.isLong;
```
Comment (EperezOk) [inline contracts/OverlayV1Market.sol:491]: No need to create a separate `isLong` variable, we can use `pos.isLong` directly (it's already loaded in memory).

**Review by EperezOk — CHANGES_REQUESTED** [review]
Last changes request, promise!

File: contracts/OverlayV1Market.sol (hunk 5 above)
Comment (magnetto90) [reply contracts/OverlayV1Market.sol]: As always, you are right!

**Review by EperezOk — APPROVED** [review]

**Review by EperezOk — APPROVED** [review]

**Pull request merged**